src/coreclr/src/System.Private.CoreLib/src/System/Threading/ProcessorIdCache.cs

```diff
@@ -0,0 +1,124 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Threading
+{
+    internal static class ProcessorIdCache
+    {
+        // The upper bits of t_currentProcessorIdCache are the currentProcessorId. The lower bits of
+        // the t_currentProcessorIdCache are counting down to get it periodically refreshed.
+        // TODO: Consider flushing the currentProcessorIdCache on Wait operations or similar
+        // actions that are likely to result in changing the executing core
+        [ThreadStatic]
+        private static int t_currentProcessorIdCache;
+
+        private const int ProcessorIdCacheShift = 16;
+        private const int ProcessorIdCacheCountDownMask = (1 << ProcessorIdCacheShift) - 1;
+        // 50 is our best guess.
+        // Based on speed check it will be adjusted, typically lower.
+        // In relatively rare cases of a slow GetCurrentProcessorNumber, it may be adjusted to a higher number.
+        private static int s_processorIdRefreshRate = 50;
```

Comment:
Looks like this is the only code in the static constructor of this type. I am not sure 100% sure about it, but it may be adding some extra overhead for `t_currentProcessorIdCache` access in some cases. It may be slightly better to leave this value uninitialized to 0 and substitute the default where it is read.

Reply:
Yeah. The initial value is now guaranteed to be overwritten. I did not think that we could avoid static constructor altogether, if we do not have this.